src/Form/ImportantSolrSettingsForm.php

```diff
@@ -0,0 +1,249 @@
+        $s = $index->getServerId();
+        if ($s === $server) {
+          $indexes_by_server[$key] = $index->label();
+        }
+      }
+      
+      return $indexes_by_server;
+  }
+
+  /**
+   * Takes a Solr Index id and return its fields
+   *
+   * @param $selected_index_id
+   *   id of the solr index
+   *
+   * @return array
+   *   array of Solr fields formatted for #options
+   */
+  public function getSolrFields($selected_index_id) {
+    // get specific Index
+    $index = $this->entityTypeManager
+      ->getStorage('search_api_index')
+      ->load($selected_index_id);
+
+    // get the fields of this Index
+    $fields = $index->get('field_settings');
+
+    // format correctly for #options
+    $fields_by_index = array();
+    foreach ($fields as $key => $field) {
+      $fields_by_index[$key] = $field['label'];
+    }
+
+    return $fields_by_index;
+  }
+  
+  /**
+   * AJAX callback function when user selects Solr Server
+   * 
+   * Updates both Index and Solr Field dropdowns in the returned AjaxResponse
+   */
+  public function onServerSelect(array &$form, FormStateInterface $form_state) {
+    // Pass the selected server to get its indexes
+    $indexes_by_server = $this->getIndexes($form_state->getValue('server_select'));
+    // As a default for this new selection, use the first index the array and fetch its fields
+    $fields = $this->getSolrFields(array_key_first($indexes_by_server));
+    
+    // Set the new #options for Index
+    $form['type']['index_select']['#options'] = $indexes_by_server;
+    // Set the new #options for Solr Field
+    $form['type']['solr_field_select']['#options'] = $fields;
+
+    // Construct our own response to return/change multiple form fields
+    $response = new AjaxResponse();
+    $response->addCommand(new ReplaceCommand("#index-select", ($form['type']['index_select'])));
+    $response->addCommand(new ReplaceCommand("#solr-field-select", ($form['type']['solr_field_select'])));
+    
+    return $response;
+   }
+
+  /**
+   * AJAX callback function when user selects Index
+   *
+   * Updates Solr Field dropdown
+   */
+  public function onIndexSelect(array &$form, FormStateInterface $form_state) {
+    // Get solr fields for selected Index
+    $fields = $this->getSolrFields($form_state->getValue('index_select'));
+    $form['type']['solr_field_select']['#options'] = $fields;
+    
+    return $form['type']['solr_field_select'];  
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('strawberryfield.archipelago_solr_settings');
+    
+    // Get all Solr servers
+    $servers = $this->entityTypeManager
+      ->getStorage('search_api_server')
+      ->loadMultiple();
+
+    // Create a fieldset for the Type setting
+    $form['type'] = array(
+      '#type' => 'fieldset',
+      '#title' => $this->t('Solr Field for Type in View Mode mapping'),
```

Comment:
This wording is not clear to me. View Mode Mapping is not defined in this module. I would say this needs to state which Solr field is going to be used as Source for an Archipelago Object's Type.

Reply:
@DiegoPino Thanks. I thinking I needed some clarification on the larger picture. That should help with other naming/labels.

Reply:
What do you think of this?

SOURCE FOR ARCHIPELAGO DIGITAL OBJECT'S TYPE

Choose the Solr field that determines an ADO's Type. Type is then used across Archipelago, for example in our ADO to View Mode mapper.

Reply:
SOURCE FOR ARCHIPELAGO DIGITAL OBJECT'S TYPE : yes!, but it needs to say also "Field" somewhere (in the description works too). And now that you mention it, the description could 😭 use a link to that index' Field config page, that way, if the field they search/want is not there, with a click they can go and add it. Also, yes, Type is then used across Archipelago, for example in our ADO to View Mode mapper, but that makes only sense if formatter_strawberryfield is enabled (i mean, why would they not have it enabled? but then people can do whatever they want eventually) so i would, maybe? remove the "for example". formatter_strawberryfield could form alter this one and add it, if enabled? What do you think? I don't want to introduce "soft" dependencies between modules.